### PR TITLE
PSP-664 Update to .NET 5.0

### PR DESCRIPTION
**Pull request by Fosol**
Updating backend API to use the latest .NET 5.0 SDK and runtime.
Requires updating the OpenShift build configuration once this has been merged.

Developers will need to rebuild backend docker container.

### CODE REVIEW

File: backend/Dockerfile.3.1 [hunk 1]
```diff
@@ -0,0 +1,40 @@
+ARG BUILD_CONFIGURATION=Release
```
Comment (Fosol) [inline backend/Dockerfile.3.1:1]: Keeping the old 3.1 around for now.

File: backend/api/Helpers/Exceptions/ApiHttpRequestException.cs [hunk 2]
```diff
@@ -10,12 +10,6 @@ namespace Pims.Api.Helpers.Exceptions
     public class ApiHttpRequestException : HttpRequestException
     {
         #region Properties
-        /// <summary>
-        /// get - The HTTP status code of the response.
-        /// </summary>
-        /// <value></value>
-        public HttpStatusCode StatusCode { get; }
```
Comment (Fosol) [inline backend/api/Helpers/Exceptions/ApiHttpRequestException.cs:17]: Fixing build warnings

File: backend/core/Http/HttpRequestClient.cs [hunk 3]
```diff
@@ -76,7 +75,7 @@ public async Task<TModel> DeserializeAsync<TModel>(HttpResponseMessage response)
             {
                 var body = Encoding.Default.GetString(data);
                 _logger.LogError(ex, $"Failed to deserialize response: {body}");
-                throw ex;
+                throw;
```
Comment (Fosol) [inline backend/core/Http/HttpRequestClient.cs:78]: Fixing build warnings

File: backend/dal/Configuration/Views/PropertyConfiguration.cs [hunk 4]
```diff
@@ -39,7 +39,7 @@ public virtual void Configure(EntityTypeBuilder<Entities.Views.Property> builder
                 // This is required to support unit-testing the view in the InMemory database.
                 builder
                     .HasNoKey()
-                    .ToQuery(() => (
+                    .ToInMemoryQuery(() => (
```
Comment (Fosol) [inline backend/dal/Configuration/Views/PropertyConfiguration.cs:42]: Replaced obsolete function

File: openshift/4.0/templates/api/build.yaml [hunk 5]
```diff
@@ -33,8 +33,7 @@ parameters:
   - name: INSTANCE
     displayName: "Unique Identifier"
     description: "A unique identifier to allow for multiple instances (i.e. '-01')."
-    required: true
```
Comment (Fosol) [inline openshift/4.0/templates/api/build.yaml:36]: This was entered incorrectly by MOTI

File: .github/workflows/api-dotnetcore.yml [hunk 6]
```diff
@@ -15,10 +15,10 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
-      - name: Setup .NET Core
+      - name: Setup .NET 5
         uses: actions/setup-dotnet@v1
         with:
-          dotnet-version: 3.1.101
+          dotnet-version: 5.0.300
```
Comment (Fosol) [inline .github/workflows/api-dotnetcore.yml:21]: Fixed GitHub action

File: backend/ches/Exceptions/ChesException.cs [hunk 7]
```diff
@@ -32,7 +32,7 @@ public class ChesException : HttpClientRequestException
         /// <param name="client"></param>
         /// <param name="model"></param>
         public ChesException(HttpClientRequestException exception, IHttpRequestClient client, ErrorResponseModel model)
-            : this($"{exception.Message}{Environment.NewLine}", exception, exception.StatusCode)
+            : base($"{exception.Message}{Environment.NewLine}", exception, exception.StatusCode ?? HttpStatusCode.InternalServerError)
```
Comment (Fosol) [inline backend/ches/Exceptions/ChesException.cs:35]: It's technically impossible to have the situation where the `exception.StatusCode == null`, but `HttpClientRequestException` inherits a nullable `StatusCode` property.

**Review by devinleighsmith — APPROVED** [review]

**Review by jdtoombs — APPROVED** [review]

**Pull request merged**